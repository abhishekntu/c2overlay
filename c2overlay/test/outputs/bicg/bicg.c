/****************************************************************/
/*   chebyshev		                                        */
/****************************************************************/

#ifdef TEST
#include <stdio.h>
#endif

#include<math.h>

#ifdef ALGO
void foo()
{
    int A [9];
    int p [3];
    int q [3];
    int r [3];
    int s [3];
    {
        q[0] = A[0] * p[0] + A[1] * p[1] + A[2] * p[2];
        q[1] = A[3] * p[0] + A[4] * p[1] + A[5] * p[2];
        q[2] = A[6] * p[0] + A[7] * p[1] + A[8] * p[2];
    }
    {
        s[0] = A[0] * r[0] + A[3] * r[1] + A[6] * r[2];
        s[1] = A[1] * r[0] + A[4] * r[1] + A[7] * r[2];
        s[2] = A[2] * r[0] + A[5] * r[1] + A[8] * r[2];
    }

}
#endif


#ifdef TEST
int main(void)
{
	int x;
 	int result;
  	foo();
  	return 0;
}
#endif
