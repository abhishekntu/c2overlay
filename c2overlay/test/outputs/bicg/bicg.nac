procedure foo()
{
  localvar s32 D_2407;
  localvar s32 D_2408;
  localvar s32 D_2409;
  localvar s32 D_2410;
  localvar s32 D_2411;
  localvar s32 D_2412;
  localvar s32 D_2413;
  localvar s32 D_2414;
  localvar s32 D_2415;
  localvar s32 D_2416;
  localvar s32 D_2417;
  localvar s32 D_2418;
  localvar s32 D_2419;
  localvar s32 D_2420;
  localvar s32 D_2421;
  localvar s32 D_2422;
  localvar s32 D_2423;
  localvar s32 D_2424;
  localvar s32 D_2425;
  localvar s32 D_2426;
  localvar s32 D_2427;
  localvar s32 D_2428;
  localvar s32 D_2429;
  localvar s32 D_2430;
  localvar s32 D_2431;
  localvar s32 D_2432;
  localvar s32 D_2433;
  localvar s32 D_2434;
  localvar s32 D_2435;
  localvar s32 D_2436;
  localvar s32 D_2437;
  localvar s32 D_2438;
  localvar s32 D_2439;
  localvar s32 D_2440;
  localvar s32 D_2441;
  localvar s32 D_2442;
  localvar s32 D_2443;
  localvar s32 D_2444;
  localvar s32 D_2445;
  localvar s32 D_2446;
  localvar s32 D_2447;
  localvar s32 D_2448;
  localvar s32 D_2449;
  localvar s32 D_2450;
  localvar s32 D_2451;
  localvar s32 A[9];
  localvar s32 p[3];
  localvar s32 q[3];
  localvar s32 r[3];
  localvar s32 s[3];
  localvar s32 t00000;
  localvar s32 t00001;
  localvar s32 t00002;
  localvar s32 t00003;
  localvar s32 t00004;
  localvar s32 t00005;
  localvar s32 t00006;
  localvar s32 t00007;
  localvar s32 t00008;
  localvar s32 t00009;
  localvar s32 t00010;
  localvar s32 t00011;
  localvar s32 t00012;
  localvar s32 t00013;
  localvar s32 t00014;
  localvar s32 t00015;
  localvar s32 t00016;
  localvar s32 t00017;
  localvar s32 t00018;
  localvar s32 t00019;
  localvar s32 t00020;
  localvar s32 t00021;
  localvar s32 t00022;
  localvar s32 t00023;
  localvar s32 t00024;
  localvar s32 t00025;
  localvar s32 t00026;
  localvar s32 t00027;
  localvar s32 t00028;
  localvar s32 t00029;
  localvar s32 t00030;
  localvar s32 t00031;
  localvar s32 t00032;
  localvar s32 t00033;
  localvar s32 t00034;
  localvar s32 t00035;
  localvar s32 t00036;
  localvar s32 t00037;
  localvar s32 t00038;
  localvar s32 t00039;
  localvar s32 t00040;
  localvar s32 t00041;
L0005:
  D_2407 <= load A,0;
  D_2408 <= load p,0;
  D_2409 <= mul D_2407,D_2408;
  D_2410 <= load A,1;
  D_2411 <= load p,1;
  D_2412 <= mul D_2410,D_2411;
  D_2413 <= add D_2409,D_2412;
  D_2414 <= load A,2;
  D_2415 <= load p,2;
  D_2416 <= mul D_2414,D_2415;
  D_2417 <= add D_2413,D_2416;
  q <= store D_2417,0;
  D_2418 <= load A,3;
  D_2408 <= load p,0;
  D_2419 <= mul D_2418,D_2408;
  D_2420 <= load A,4;
  D_2411 <= load p,1;
  D_2421 <= mul D_2420,D_2411;
  D_2422 <= add D_2419,D_2421;
  D_2423 <= load A,5;
  D_2415 <= load p,2;
  D_2424 <= mul D_2423,D_2415;
  D_2425 <= add D_2422,D_2424;
  q <= store D_2425,1;
  D_2426 <= load A,6;
  D_2408 <= load p,0;
  D_2427 <= mul D_2426,D_2408;
  D_2428 <= load A,7;
  D_2411 <= load p,1;
  D_2429 <= mul D_2428,D_2411;
  D_2430 <= add D_2427,D_2429;
  D_2431 <= load A,8;
  D_2415 <= load p,2;
  D_2432 <= mul D_2431,D_2415;
  D_2433 <= add D_2430,D_2432;
  q <= store D_2433,2;
  D_2407 <= load A,0;
  D_2434 <= load r,0;
  D_2435 <= mul D_2407,D_2434;
  D_2418 <= load A,3;
  D_2436 <= load r,1;
  D_2437 <= mul D_2418,D_2436;
  D_2438 <= add D_2435,D_2437;
  D_2426 <= load A,6;
  D_2439 <= load r,2;
  D_2440 <= mul D_2426,D_2439;
  D_2441 <= add D_2438,D_2440;
  s <= store D_2441,0;
  D_2410 <= load A,1;
  D_2434 <= load r,0;
  D_2442 <= mul D_2410,D_2434;
  D_2420 <= load A,4;
  D_2436 <= load r,1;
  D_2443 <= mul D_2420,D_2436;
  D_2444 <= add D_2442,D_2443;
  D_2428 <= load A,7;
  D_2439 <= load r,2;
  D_2445 <= mul D_2428,D_2439;
  D_2446 <= add D_2444,D_2445;
  s <= store D_2446,1;
  D_2414 <= load A,2;
  D_2434 <= load r,0;
  D_2447 <= mul D_2414,D_2434;
  D_2423 <= load A,5;
  D_2436 <= load r,1;
  D_2448 <= mul D_2423,D_2436;
  D_2449 <= add D_2447,D_2448;
  D_2431 <= load A,8;
  D_2439 <= load r,2;
  D_2450 <= mul D_2431,D_2439;
  D_2451 <= add D_2449,D_2450;
  s <= store D_2451,2;
  L_EXIT <= jmpun;
L_EXIT:
  nop;
}
