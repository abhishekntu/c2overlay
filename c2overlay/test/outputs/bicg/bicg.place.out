Netlist file: test.net   Architecture file: test.arch.xml
Array size: 7 x 7 logic blocks

#block name	x	y	subblk	block number
#----------	--	--	------	------------
N8		3	8	0	#0
N9		3	0	0	#1
N1		0	4	0	#2
N2		0	2	0	#3
N3		2	0	0	#4
N4		8	3	0	#5
N5		8	5	0	#6
N6		5	8	0	#7
N7		6	0	0	#8
N12		4	8	0	#9
N13		1	0	0	#10
N10		0	3	0	#11
N11		8	2	0	#12
N14		0	6	0	#13
N15		4	0	0	#14
out:N38		7	0	0	#15
out:N39		5	0	0	#16
out:N34		8	1	0	#17
out:N35		8	4	0	#18
out:N36		0	1	0	#19
out:N37		0	5	0	#20
N29_blk		3	3	0	#21
N28_blk		2	1	0	#22
N23_blk		2	3	0	#23
N22_blk		6	2	0	#24
N21_blk		5	2	0	#25
N20_blk		1	1	0	#26
N27_blk		7	2	0	#27
N26_blk		5	1	0	#28
N25_blk		4	2	0	#29
N24_blk		3	2	0	#30
N16_blk		3	1	0	#31
N17_blk		2	2	0	#32
N18_blk		4	1	0	#33
N19_blk		1	3	0	#34
N30_blk		7	1	0	#35
N31_blk		6	1	0	#36
N32_blk		4	3	0	#37
N33_blk		1	2	0	#38
